### PR TITLE
Track the absolute time that presence events are received, so that the relative lastActiveAgo value is meaningful.

### PR DESCRIPTION
**Pull request by ara4n**
This solves SYJS-28

### CODE REVIEW

File: lib/models/room.js [hunk 1]
```diff
@@ -695,6 +695,16 @@ Room.prototype._addLiveEvent = function(event, duplicateStrategy) {
...
+
+        // also, any live events from a user should be taken as implicit
+        // presence information: evidence that they are currently active.
+        // ...except in a world where we use 'user.currentlyActive' to reduce
+        // presence spam, this isn't very useful - we'll get a transition when
+        // they are no longer currently active anyway. so comment it out for now.
+
+        // var user = this.currentState.getMember(events[i].sender.userId);
+        // user.lastActiveAgo = 0;
+        // user.lastPresenceTs = Date.now();
```
Comment (Kegsay) [inline lib/models/room.js]: If we don't plan to ever reinstate this code, please delete it.

File: lib/models/user.js [hunk 2]
```diff
@@ -30,7 +30,12 @@ limitations under the License.
  * @prop {string} displayName The 'displayname' of the user if known.
  * @prop {string} avatarUrl The 'avatar_url' of the user if known.
  * @prop {string} presence The presence enum if known.
- * @prop {Number} lastActiveAgo The last time the user performed some action in ms.
+ * @prop {Number} lastActiveAgo The time elapsed in ms since the user interacted
+ *                proactively with the server, or we saw a message from the user
+ * @prop {Number} lastPresenceTs Timestamp (ms since the epoch) for when we last
+ *                received presence data for this user.  We can subtract
+ *                lastActiveAgo from this to approximate an absolute value for
+ *                when a user was last active.
```
Comment (Kegsay) [inline lib/models/user.js]: It would be useful to have a method on `User` to do this calculation rather than just mentioning it in the docs.

**Comment by Kegsay** [comment]
LGTM aside from comments.

**Comment by ara4n** [comment]
@Kegsay PTAL

**Comment by ara4n** [comment]
merging as i want to ship this and believe i addressed K's comments

**Pull request merged**